kubernetes/update.go

```diff
@@ -141,7 +141,19 @@ func updateDeployment(ctx context.Context, deployCli v1.DeploymentInterface, dep
 		Str("deploymentName", deploymentName).
 		Msg("Waiting for deployment to complete")
 
-	return waitForDeployment(ctx, deployCli, newDeployment.Name, newDeployment.UID)
+	err = nil
+	numRetries := 1
+	for numRetries < 10 {
+		err = waitForDeployment(ctx, deployCli, newDeployment.Name, newDeployment.UID)
+		if err == nil {
+			return nil
+		}
+
+		log.Warn().Err(err).Int("number of retries", numRetries).Msg("Deployment not ready yet, retrying")
+		numRetries++
+	}
+
+	return errors.WithMessage(err, "Something went wrong, deployment not ready after 10 retries")
```

Comment:
"Something went wrong" is not a very good error message.  

Also "github.com/pkg/errors"  should not be used.